figures/tasks.py

```diff
@@ -328,6 +374,63 @@ def populate_daily_metrics_next(site_id=None, force_update=False):
                            site_count=sites_count))
 
 
+@shared_task
+def daily_metrics_callback(results, *args):
+    logger.info(results)
+    logger.info(args)
+
+
+@shared_task
+def populate_daily_metrics_async_sites(site_id=None, force_update=False):
+    """Next iteration to collect daily metrics for all sites in a deployment
+
+    This is a top level Celery task run every 24 hours to update Figures data.
+
+    * It updates Figures per-enrollment data and collect daily aggregate metrics
+    * It's purpose is to collect new metrics on an ongoing basis and not serve
+      dual purpose of collecting ongoing data AND backfilling data.
+    * The driver for this change is to improve performance of the daily Celery jobs
+
+    What's different?
+
+    * Figures collects the enrollment data first, then aggregates daily data.
+
+    TODO: Draft up public architecture docs and reference them here
+    """
+    if waffle.switch_is_active(WAFFLE_DISABLE_PIPELINE):
+        logger.warning('Figures pipeline is disabled due to %s being active.',
+                       WAFFLE_DISABLE_PIPELINE)
+        return
+
+    date_for = datetime.datetime.utcnow().date()
+    if site_id is not None:
+        sites = get_sites_by_id((site_id, ))
+    else:
+        sites = get_sites()
+    sites_count = sites.count()
+    # This is our task entry log message
+    msg = '{prefix}:START:date_for={date_for}, site_count={site_count}'
+    logger.info(msg.format(prefix=FPD_LOG_PREFIX,
+                           date_for=date_for,
+                           site_count=sites_count))
+
+    logger.info('Starting figures.tasks.populate_daily_metrics...')
+    msg_metrics_end = '{prefix}:END:date_for={date_for}, site_count={site_count}'.format(
+        prefix=FPD_LOG_PREFIX, date_for=date_for, site_count=sites_count)
+
+    daily_tasks_list = [
+        populate_daily_metrics_for_site_async_sites.s(
+            site_id=site.id,
+            date_for=date_for.strftime('%m/%d/%Y'),
```

Comment:
I find it a bit odd that we're serializing `date_for` into a string to pass to other functions. It seems like it would be better to keep it a `datetime` object and just make the rest of the code work that way. Doing it this way just seems to be asking for bugs with timezones, etc.